src/Routing/Router.php

```diff
@@ -569,9 +569,9 @@ public static function url($url = null, $full = false)
         }
 
         if (empty($url)) {
-            $output = isset($here) ? $here : '/';
+            $output = isset($here) ? $here : $base . '/';
             if ($full) {
-                $output = static::fullBaseUrl() . $base . $output;
+                $output = static::fullBaseUrl() . $output;
```

Comment:
Is there a way we could reproduce the original issue in a test case. This helps us avoid regressions in the future.